### PR TITLE
account for a 2nd layer in generated mlbf in test_generate_and_write_mlbf

### LINKED ISSUE #14452: Intermittent test failure in blocklist/tests/test_mlbf.py: test_generate_and_write_mlbf()
We've seen this a few times:
```
____________________ TestMLBF.test_generate_and_write_mlbf _____________________
[gw0] linux -- Python 3.6.7 /home/travis/build/mozilla/addons-server/.tox/main/bin/python3
src/olympia/blocklist/tests/test_mlbf.py:274: in test_generate_and_write_mlbf
    assert bfilter.bitCount() == 1384
E   AssertionError: assert 2824 == 1384
E     +2824
E     -1384
        bfilter    = <filtercascade.FilterCascade object at 0x7f5731cec358>
        buffer     = (b'\x02\x00\x00\x10\xaa\x8f^\\r4\xf9\x9a%%\xb4\x0eI\x1e\xdf\x89\x02h\x05\x00'
 b'\x00\x01\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x80\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x10\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 \x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x02\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 \x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x04\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x02\xa0\x05\x00\x00\x01\x00\x00\x00\x02\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x02\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'
 b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00')
        filter_file = <_io.BufferedReader name='/tmp/pytest-of-travis/pytest-0/popen-gw0/test_generate_and_write_mlbf0/storage/mlbf/123456/filter'>
        mlbf       = <olympia.blocklist.mlbf.MLBF object at 0x7f57316b94a8>
        self       = <olympia.blocklist.tests.test_mlbf.TestMLBF testMethod=test_generate_and_write_mlbf>
```

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #14452

### CODE REVIEW

File: src/olympia/blocklist/tests/test_mlbf.py [hunk 1]
```diff
@@ -288,9 +288,15 @@ def test_generate_and_write_mlbf(self):
             key = mlbf.KEY_FORMAT.format(guid=guid, version=version_str)
             assert key not in bfilter
 
-        assert os.stat(mlbf.filter_path).st_size == 203, (
+        # Occasionally a combination of salt generated with secrets.token_bytes
+        # and the version str generated in version_factory results in a
+        # collision in layer 1 of the bloomfilter, leading to a second layer
+        # being generated.  When this happns the bitCount and size is larger.
```
Comment (willdurand) [inline src/olympia/blocklist/tests/test_mlbf.py:294]:
```suggestion
        # being generated.  When this happens the bitCount and size is larger.
```

File: src/olympia/blocklist/tests/test_mlbf.py [hunk 2]
```diff
@@ -288,9 +288,15 @@ def test_generate_and_write_mlbf(self):
             key = mlbf.KEY_FORMAT.format(guid=guid, version=version_str)
             assert key not in bfilter
 
-        assert os.stat(mlbf.filter_path).st_size == 203, (
+        # Occasionally a combination of salt generated with secrets.token_bytes
+        # and the version str generated in version_factory results in a
```
Comment (willdurand) [inline src/olympia/blocklist/tests/test_mlbf.py:292]: Can't we set a fixed version str?

Comment (eviljeff) [reply src/olympia/blocklist/tests/test_mlbf.py:292]: we could but that wouldn't fix it.  The issue has more detail if the comment isn't clear.

Comment (willdurand) [reply src/olympia/blocklist/tests/test_mlbf.py:292]: ha, ok. I guess we could mock the randomness to avoid the issue then?

**Pull request merged**